File: identify/src/main/java/com/lytmoon/identify/CharCenter.java
package com.lytmoon.identify;

import com.google.gson.annotations.SerializedName;
import com.google.gson.Gson;
import java.util.Base64;
import java.util.List;

class CharCenter {
    @SerializedName("x")
    public int x;

    @SerializedName("y")
    public int y;

    // Getters and setters
}

class CharPolygon {
    @SerializedName("x1")
    public int x1;

    @SerializedName("y1")
    public int y1;

    @SerializedName("x2")
    public int x2;

    @SerializedName("y2")
    public int y2;

    @SerializedName("x3")
    public int x3;

    @SerializedName("y3")
    public int y3;

    @SerializedName("x4")
    public int x4;

    @SerializedName("y4")
    public int y4;

    // Getters and setters
}

class Line {
    @SerializedName("angle")
    public int angle;

    @SerializedName("char_centers")
    public List<List<Integer>> charCenters;

    @SerializedName("char_polygons")
    public List<List<Integer>> charPolygons;

    @SerializedName("char_score")
    public List<Double> charScore;

    @SerializedName("position")
    public List<Integer> position;

    @SerializedName("property")
    public int property;

    @SerializedName("score")
    public double score;

    @SerializedName("text")
    public String text;

    // Getters and setters
}

class DataClass {
    @SerializedName("image_angle")
    public int imageAngle;

    @SerializedName("lines")
    public List<Line> lines;

    @SerializedName("property_map")
    public List<String> propertyMap;

    @SerializedName("rotated_image_height")
    public int rotatedImageHeight;

    @SerializedName("rotated_image_width")
    public int rotatedImageWidth;

    @SerializedName("whole_text")
    public String wholeText;


}

File: lib_api/src/main/java/com/example/api/orc/dataBean/CharCenter.kt
package com.example.api.orc.dataBean

import com.google.gson.annotations.SerializedName

data class CharCenter(
    @SerializedName("x") val x: Int,
    @SerializedName("y") val y: Int
)

data class CharPolygon(
    @SerializedName("x1") val x1: Int,
    @SerializedName("y1") val y1: Int,
    @SerializedName("x2") val x2: Int,
    @SerializedName("y2") val y2: Int,
    @SerializedName("x3") val x3: Int,
    @SerializedName("y3") val y3: Int,
    @SerializedName("x4") val x4: Int,
    @SerializedName("y4") val y4: Int
)

data class Line(
    @SerializedName("angle") val angle: Int,
    @SerializedName("char_centers") val charCenters: List<List<Int>>,
    @SerializedName("char_polygons") val charPolygons: List<List<Int>>,
    @SerializedName("char_score") val charScore: List<Double>,
    @SerializedName("position") val position: List<Int>,
    @SerializedName("property") val property: Int,
    @SerializedName("score") val score: Double,
    @SerializedName("text") val text: String
)

data class DataClass(
    @SerializedName("image_angle") val imageAngle: Int,
    @SerializedName("lines") val lines: List<Line>,
    @SerializedName("property_map") val propertyMap: List<String>,
    @SerializedName("rotated_image_height") val rotatedImageHeight: Int,
    @SerializedName("rotated_image_width") val rotatedImageWidth: Int,
    @SerializedName("whole_text") val wholeText: String
)
